src/intervals/flavors.jl

```diff
@@ -0,0 +1,80 @@
+"""
```

Comment:
Suggestion: This file can/should be split into two: one which contains all the common stuff required for any flavor, and other where the set_based flavor is implemented. Then, the second file can be used to construct the infrastructure of other flavors.